go/client/cmd_team_edit.go

```diff
@@ -0,0 +1,170 @@
+package client
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/keybase/cli"
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+	"golang.org/x/net/context"
+)
+
+type CmdTeamEdit struct {
+	libkb.Contextified
+	Team      keybase1.TeamName
+	PrintOnly bool
+	Settings  keybase1.TeamSettings
+}
+
+func newCmdTeamEdit(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:         "edit",
+		ArgumentHelp: "<team name>",
+		Usage:        "Edit team settings.",
+		Action: func(c *cli.Context) {
+			cmd := NewCmdTeamEditRunner(g)
+			cl.ChooseCommand(cmd, "edit", c)
+		},
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "open",
+				Usage: "Change team to be open or closed",
+			},
+			cli.StringFlag{
+				Name:  "join-as",
+				Usage: "team role (writer, reader) [required when changing team to open]",
+			},
+		},
+	}
+}
+
+func NewCmdTeamEditRunner(g *libkb.GlobalContext) *CmdTeamEdit {
+	return &CmdTeamEdit{Contextified: libkb.NewContextified(g)}
+}
+
+func (c *CmdTeamEdit) ParseArgv(ctx *cli.Context) error {
+	var err error
+	c.Team, err = ParseOneTeamNameK1(ctx)
+	if err != nil {
+		return err
+	}
+
+	if ctx.NumFlags() == 0 {
+		// Just a team name and no other flags - print current team settings.
+		c.PrintOnly = true
+		return nil
+	}
+
+	sopen := ctx.String("open")
+	switch sopen {
+	case "true":
+		c.Settings.Open = true
+	case "false":
+		c.Settings.Open = false
+	default:
+		return errors.New("invalid --open flag, please use true or false")
+	}
+
+	if c.Settings.Open {
+		srole := ctx.String("join-as")
+		if srole == "" {
+			return errors.New("team role required via --join-as flag")
+		}
+
+		role, ok := keybase1.TeamRoleMap[strings.ToUpper(srole)]
+		if !ok {
+			return errors.New("invalid team role, please use writer, or reader")
+		}
+
+		switch role {
+		case keybase1.TeamRole_READER, keybase1.TeamRole_WRITER:
+			break
+		default:
+			return errors.New("invalid team role, please use writer, or reader")
+		}
+
+		c.Settings.JoinAs = role
+	}
+
+	return nil
+}
+
+func (c *CmdTeamEdit) applySettings(cli keybase1.TeamsClient) error {
+	dui := c.G().UI.GetTerminalUI()
+
+	details, err := cli.TeamGet(context.Background(), keybase1.TeamGetArg{Name: c.Team.String(), ForceRepoll: false})
+	if err != nil {
+		return err
+	}
+
+	if !details.Settings.Open && !c.Settings.Open {
+		dui.Printf("The team is already closed.\n")
+		return nil
+	}
+
+	if details.Settings.Open && c.Settings.Open && details.Settings.JoinAs == c.Settings.JoinAs {
+		dui.Printf("The team is already open with default role: %s\n", strings.ToLower(details.Settings.JoinAs.String()))
+		return nil
+	}
```

Comment:
I think you should move this logic into the service `TeamSetSettings` handler so that it is there for GUI client as well.